doc/man7/EVP_PKEY-EC.pod

```diff
@@ -79,6 +79,26 @@ EC curve's cofactor (note for some curves the cofactor is 1).
 Set the format used for serializing the EC group parameters.
 Valid values are "explicit" or "named_curve". The default value is "named_curve".
 
+=item "point-format" (B<OSSL_PKEY_PARAM_EC_POINT_CONVERSION_FORMAT>) <utf8 string>
+
+Sets or gets the point_conversion_form for the I<key>. For a description of
+point_conversion_forms please see L<EC_POINT_new(3)>. Valid values are
+"uncompressed" or "compressed". The default value is "uncompressed".
+
+=item "group-check" (B<OSSL_PKEY_PARAM_EC_GROUP_CHECK_TYPE>) <utf8 string>
+
+Sets or Gets the type of group check done when EVP_PKEY_param_check() is called.
+Valid values are "default", "named" and "named-nist".
+The "named" type checks that the domain parameters match the inbuilt curve parameters,
+"named-nist" is similiar but also checks that the named curve is a nist curve.
+The "default" type does domain parameter validation for the OpenSSL default provider,
+but is equivalent to "named-nist" for the OpenSSL fips provider.
+
+=item "publickey-enabled" (B<OSSL_PKEY_PARAM_EC_PUBKEY_ENABLED>) <integer>
+
+Setting this value to 0 indicates that the public key should not be included when
+encoding the private key. The default value of 1 will include the public key.
```

Comment:
I've always found this to be a *weird* thing to set in the key itself.  This should really be an encoder parameters, shouldn't it?

Reply:
Yes, and I believe the point-format parameter is also such thing in a sense as it is a property of the representation of the key when encoded but it has no effect on the internal data representation if the key is supposed to be used in operations.
And the group-check is not a property of a key either. It is a parameter of the EVP_PKEY_param_check() operation so it should be set on the EVP_PKEY_CTX not on the key itself.

Reply:
One issue with the point-format parameter is that it might be useful to record it when the key is decoded. So I can see it could be useful to have it as read-only parameter on the key. But you should be able to override it when encoding.

Reply:
Something to be noted is that this weirdness may be driven by external "forces"...  EC keys are surrounded with quite a bit of weirdness, due to the iterations in their development over time.

Reply:
Left for now as is. Although I do not like it much, it is nothing major.